libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -833,16 +854,31 @@ export abstract class CommandLineParameterProvider {
           throw new Error('Unexpected parameter group: ' + parameterGroup);
         }
 
-        argumentGroup = this._getArgumentParser().addArgumentGroup({
+        argumentGroup = argumentParser.addArgumentGroup({
           title: `Optional ${parameterGroupName} arguments`
         });
         this._parameterGroupsByName.set(parameterGroup, argumentGroup);
       }
     } else {
-      argumentGroup = this._getArgumentParser();
+      argumentGroup = argumentParser;
     }
 
-    argumentGroup.addArgument(names, { ...argparseOptions });
+    const argparseArgument: argparse.ArgumentOptions = (argumentGroup as IExtendedArgumentGroup).addArgument(
+      names,
+      argparseOptions
+    );
+    if (required && environmentVariable) {
+      parameter._preParse = () => {
+        argparseArgument.required = false;
+      };
+      parameter._postParseValidation = (value: unknown | null | undefined) => {
```

Comment:
Seems like an odd place to specify this... could we specify the pre/postParse on the base parameter instead? If not, this will work, but it does seem odd to be touching the parameter after registration

Reply:
It only needs to do something in this particular case (env var+required), and it needs the argparse argument object, so defining it in the base class will be inconvenient and slower.